x-pack/plugins/infra/server/routes/infra/lib/mapper.ts

```diff
@@ -32,16 +34,29 @@ export const mapToApiResponse = (
     };
   }
 
-  const hosts = buckets.map((bucket) => {
-    const metrics = convertMetricBucket(params, bucket);
-    const metadata = convertMetadataBucket(bucket);
+  const hosts = buckets
+    .map((bucket) => {
+      const metrics = convertMetricBucket(params, bucket);
+      const metadata = convertMetadataBucket(bucket);
 
-    return { name: bucket.key as string, metrics, metadata };
-  });
+      return { name: bucket.key as string, metrics, metadata };
+    })
+    .sort((a, b) => {
+      const cpuAValue = a.metrics.find((metric) => metric.name === 'cpu')?.value ?? null;
+      const cpuBValue = b.metrics.find((metric) => metric.name === 'cpu')?.value ?? null;
+
+      // If both cpuAValue and cpuBValue are not null or undefined, perform the comparison
+      return (cpuBValue! ?? 0) - (cpuAValue! ?? 0);
+    });
+
+  const mergedHostWithAlertsCount = hosts.map((itm) => ({
+    ...alertsCountResponse?.find((item) => item.name === itm.name),
```

Comment:
It might not be a problem, but at this moment you have a O(n^2) time complexity. Maybe we you could do is to group the `alertsCountResponse` based on the `name` property. Then here you could just do:
```suggestion
    ...alertsCountGroupedByName[itm.name],
```

Reply:
I moved the merging to the first map as Carlos proposal here https://github.com/elastic/kibana/pull/176034#discussion_r1482641732